commit bcb82414c2f69e015db27d699f27b51ca3b04add
Author: coldrunKacper <kacper@coldrun.pl>
Date:   Tue Jun 16 14:02:53 2015 +0200

    Adding all extensions path to include_path inside CRM/Core/Config.php::_initialize method and removing include_path duplicates inside CRM/Utils/Hook.php::config hook.
    
    Setting all extensions paths to include_path.

diff --git a/CRM/Core/Config.php b/CRM/Core/Config.php
index 72553ed..ac08246 100644
--- a/CRM/Core/Config.php
+++ b/CRM/Core/Config.php
@@ -398,6 +398,20 @@ class CRM_Core_Config extends CRM_Core_Config_Variables {
     // between CRM_Core_Config and CRM_Core_Component (they
     // are co-dependant)
     $this->componentRegistry = new CRM_Core_Component();
+    
+    // add active extensions path to include_path
+    // previously it was added later for particular extension
+    // in each extension.civix.php script
+    // but doing this earlier allows to use components inside extensions
+    global $civicrm_root;
+    $extensionsContainer = new CRM_Extension_Container_Basic($civicrm_root, $this->resourceBase);
+    $extensions = $extensionsContainer->getKeys();
+    $enableExtensions = array();
+    foreach ($extensions as $extension)
+    {
+        $enableExtensions[$extension] = $extensionsContainer->getPath($extension) . '/';
+    }
+    set_include_path(implode(PATH_SEPARATOR, $enableExtensions) . PATH_SEPARATOR . get_include_path());
   }
 
   /**
diff --git a/CRM/Utils/Hook.php b/CRM/Utils/Hook.php
index 3ca5709..e3c1f84 100644
--- a/CRM/Utils/Hook.php
+++ b/CRM/Utils/Hook.php
@@ -901,10 +901,12 @@ abstract class CRM_Utils_Hook {
    * @return mixed
    */
   static function config(&$config) {
-    return self::singleton()->invoke(1, $config,
+    $invoked = self::singleton()->invoke(1, $config,
       self::$_nullObject, self::$_nullObject, self::$_nullObject, self::$_nullObject, self::$_nullObject,
       'civicrm_config'
     );
+    set_include_path(implode(PATH_SEPARATOR, array_unique(explode(PATH_SEPARATOR, get_include_path()))));
+    return $invoked;
   }
 
   /**
